exchangelib/services.py

```diff
@@ -511,6 +511,19 @@ def _get_elements_in_response(self, response):
     def _get_elements_in_container(container):
         return [elem for elem in container]
 
+    def add_additional_properties_to_shape(self, shape_element, additional_fields):
+        if not additional_fields:
+            return
+
+        from .fields import FieldPath
+        if not isinstance(list(additional_fields)[0], FieldPath):
```

Comment:
Is additional fields a list of fields to be included? Is this only checking if the first element is of type `FieldPath`?

Reply:
Yeah, originally only lists of `FieldPath`s were passed in, but I wanted to abstract that out a bit, and assumed that if the first element wasn't a `FieldPath`, then none of them would be. If we wanted to be more robust though, I could change this to be
```
additional_field_paths = []
for field in additional_fields:
    if isinstance(field, FieldPath):
        field_path = field
    else:
        field_path = FieldPath(field=field)
    additional_field_paths.append(field_path)
```

Reply:
Gotcha I think I like this way better